### PR TITLE
initial testing checklist

### PR DESCRIPTION
**Pull request by scholarsmate**
Provides a checklist of things to check before and during a release cycle.

### CODE REVIEW

File: src/tests/checklists/test-checklist.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+## DFDL Debugger
+- [ ] open a DFDL file
+- [ ] verify source highlighting
+- [ ] verify code completion
+- [ ] more here...
+
+## TDML
+- [ ] open a TDML file
+- [ ] Generate a TDML file from debugging a daffodil schema
```
Comment (scholarsmate) [inline src/tests/checklists/test-checklist.md:42]: start these in lower case to be consistent with the others.

File: src/tests/checklists/test-checklist.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+- [ ] Verify that the generated TDML file contains correct relative paths
+- [ ] Veirfy parserTestCase object has name, root, model, description
```
Comment (scholarsmate) [inline src/tests/checklists/test-checklist.md:44]: Veirfy => verify

File: src/tests/checklists/test-checklist.md
Comment (arosien) [inline src/tests/checklists/test-checklist.md:1]: Suggestion: name this `src/tests/README.md`.

Comment (scholarsmate) [reply src/tests/checklists/test-checklist.md:1]: I'm okay with that.

Comment (scholarsmate) [reply src/tests/checklists/test-checklist.md:1]: @NolanMatt, can you rename the file in addition to the other requested changes, then re-push?

File: src/tests/README.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+  - [ ] scroll through the file, using the pseudo scrollbar
+  - [ ] scroll through the file, using the mouse wheel
+  - [ ] scroll through the file, using keyboard shortcuts (up, down, page up, page down, home, end)
+- [ ] display modes
+  - [ ] change the address radix (hexadecimal, decimal, octal)
+  - [ ] change the display radix (hexadecimal, decimal, octal, binary)
+- [ ] profiler
+  - [ ] profile the file using the profile button
+  - [ ] toggle the overlay (ASCII, None)
+  - [ ] change the start offset (use legal, illegal values, negative values, values that are too large, etc.)
```
Comment (stricklandrbls) [inline src/tests/README.md:71]: Changed the start offset to `FF` and it resulted the 'Start Offset' and 'Length' being set to `NaN`.
<details><summary><h3>Screenshot</h3></summary>

image

</details>

File: src/tests/README.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+  - [ ] verify that the header min/max button shows and hides the header
+- [ ] navigation
+  - [ ] scroll through the file, using navigation buttons
+  - [ ] scroll through the file, using the pseudo scrollbar
+  - [ ] scroll through the file, using the mouse wheel
+  - [ ] scroll through the file, using keyboard shortcuts (up, down, page up, page down, home, end)
+- [ ] display modes
+  - [ ] change the address radix (hexadecimal, decimal, octal)
+  - [ ] change the display radix (hexadecimal, decimal, octal, binary)
+- [ ] profiler
```
Comment (stricklandrbls) [inline src/tests/README.md:68]: ## Profiler Discrepancies
### Offsets and Length Changes are `NaN`
Changed the start/end offset to `FF` and it resulted the 'Start/End Offset' and 'Length' being set to `NaN`.
<details><summary><h4>Screenshot</h4></summary>

image

</details>

---

### Assert Failure on Length Change
A console error in regards to an assertion failure appears when changing the value of the 'Length' field.
<details><summary><h4>Screenshot</h4></summary>

image

</details>

**Review by NolanMatt — APPROVED** [review]
Worked very well for testing for 1.3.1

**Review by arosien — APPROVED** [review]
👍

**Review by stricklandrbls — APPROVED** [review]
+1

**Pull request merged**